pywr/notebook/__init__.py

```diff
@@ -57,23 +69,76 @@ def pywr_model_to_d3_json(model):
             else:
                 cls = classes[-1]
         classes = classes[::-1]
-        node["clss"] = [cls.__name__.lower() for cls in classes]
+        node_dict["clss"] = [cls.__name__.lower() for cls in classes]
         try:
-            node["position"] = node_obj.position["schematic"]
+            node["position"] = node.position["schematic"]
         except KeyError:
             pass
-        json_nodes.append(node)
+    
+        if attributes:
+            node_dict["attributes"] = get_node_attr(node)
+        
+        json_nodes.append(node_dict)
 
     graph = {
         "nodes": json_nodes,
         "links": edges}
 
+
     return graph
 
+def get_node_attr(node):
+    """
+    Returns a dictionary that contains node attributes as strings
 
-def pywr_json_to_d3_json(model):
+    Parameters
+    ----------
+    node : a pywr node object 
     """
-    converts a json file or a json-derived dict into structure that D3 can use
+    attrs = inspect.getmembers(node, lambda a:not(inspect.isroutine(a)))
+    attribute_data = []
+    for att in attrs:
+        
+        attr_name, attr_val = att
+        if attr_name.startswith("_"):
+            continue
+        attr_type = type(attr_val).__name__ 
+        
+        attrs_to_skip = ["component_attrs", "components", "color", "model",  "input", "output",
```

Comment:
What attributes are we trying to get here? This seems a bit hackish on a second look. I guess it is related to the schema stuff? Perhaps if that is merged the schema should be used here to dump nodes to JSON data.